src/lake/Lake/Build/Module.lean

```diff
@@ -169,13 +172,30 @@ def Module.cFacetConfig : ModuleFacetConfig cFacet :=
       -- do content-aware hashing so that we avoid recompiling unchanged C files
       return (mod.cFile, ← fetchFileTrace mod.cFile)
 
+/-- The `ModuleFacetConfig` for the builtin `bcFacet`. -/
+def Module.bcFacetConfig : ModuleFacetConfig bcFacet :=
+  mkFacetJobConfigSmall fun mod => do
+    (← mod.leanArts.fetch).bindSync fun _ _ =>
+      -- do content-aware hashing so that we avoid recompiling unchanged bitcode files
+      return (mod.bcFile, ← fetchFileTrace mod.bcFile)
+
 /-- Recursively build the module's object file from its C file produced by `lean`. -/
-def Module.recBuildLeanO (self : Module) : IndexBuildM (BuildJob FilePath) := do
-  buildLeanO self.name.toString self.oFile (← self.c.fetch) self.leancArgs
+def Module.recBuildLeanCToO (self : Module) : IndexBuildM (BuildJob FilePath) := do
+  buildLeanFromCToO self.name.toString self.oFile (← self.c.fetch) self.leancArgs
+
+/-- Recursively build the module's object file from its bitcode file produced by `lean`. -/
+def Module.recBuildLeanBcToO (self : Module) : IndexBuildM (BuildJob FilePath) := do
+  -- TODO: add option to pass target.
+  buildLeanFromBcToO self.name.toString self.oFile (← self.bc.fetch)
 
 /-- The `ModuleFacetConfig` for the builtin `oFacet`. -/
 def Module.oFacetConfig : ModuleFacetConfig oFacet :=
-  mkFacetJobConfig Module.recBuildLeanO
+  -- TODO: add option to switch between `.bc` and `.o` builds.
+  mkFacetJobConfig (fun module => 
+    if module.backend == Backend.C
```

Comment:
Can we do a match here instead in case we want to add other targets later? Otherwise we would probably get confused about why it is trying to do LLVM stuff per default for a while